### DISCUSSION TITLE
Eslint does not show error in command line for all js files in the project

### DISCUSSION THREAD

**Discussion by JohnCasab** [eslint/eslint#13932] [Q&A]
Hello Eslint community,

**Tell us about your environment**
```
Node version: v15.4.0
NPM version: 7.0.15
Yarn version: 1.22.10
Eslint local version: 7.2.0

```
**Configuration**
```
{
  "name": "html-storybook",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook",
    "lint": "./node_modules/.bin/eslint '**/*.js'"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "bootstrap": "^4.5.3",
    "html-react-parser": "^0.14.0",
    "jquery": "^3.5.1",
    "popper.js": "^1.16.1",
    "sass": "^1.29.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@storybook/addon-actions": "^6.1.9",
    "@storybook/addon-docs": "^6.1.9",
    "@storybook/addon-essentials": "^6.1.9",
    "@storybook/addon-links": "^6.1.9",
    "@storybook/html": "^6.1.9",
    "@storybook/preset-scss": "^1.0.3",
    "babel-loader": "^8.1.0",
    "css-loader": "^5.0.1",
    "eslint": "^7.2.0",
    "eslint-config-airbnb-base": "^14.2.1",
    "eslint-plugin-import": "^2.22.1",
    "react-is": "^16.13.1",
    "sass-loader": "^10.1.0",
    "storybook-addon-run-script": "^0.1.4",
    "style-loader": "^2.0.0"
  }
}
```

eslintrc.json:
```
{
    "env": {
        "node": true,
        "browser": true,
        "es2020": true,
        "commonjs": true,
        "jquery": true
    },
    "extends": [
        "airbnb-base"
    ],
    "parserOptions": {
        "ecmaVersion": 2020,
        "sourceType": "module"
    },
    "rules": {
    }
}
```

.eslintignore:
`!/.storybook`

**The problem**
When I run `yarn lint`, I only see my lint errors for the js files in my src folder. Like this: 
<img width="1156" alt="picture" src="https://user-images.githubusercontent.com/72789740/102352603-3fa89300-3fa8-11eb-8e2c-00d8e4213468.png">

I created an `test.js` file in my .storybook folder, and created it with errors but I dont see them, when I run `yarn lint`, so I thought maybe its only in the folders with dot prefix, but I also created `test.js` in `helm-chart` folder but that is also not detected when I run `yarn lint`. 

When i run the command `yarn run lint --fix` it actually fixes all my lint problems in both of my `test.js` files in `.storybook `and `helm-chart` folders. 

**What did you expect to happen?**
I expect to see all lint errors, in all of my .js files in my project.

**Comment by mdjermanovic** [comment]
Hi @JohnCasab, thanks for the issue!

> When i run the command `yarn run lint --fix` it actually fixes all my lint problems in both of my `test.js` files in `.storybook `and `helm-chart` folders.

It's weird that `--fix` fixes the problems but the same command without `--fix` doesn't show them.

Can you run linting with the  [--debug](https://eslint.org/docs/user-guide/command-line-interface#-debug) flag and share the output?

**Comment by JohnCasab** [comment]
Hi @mdjermanovic 

I actually solved the problem, what I did was in my `.eslintrc.json` file changed the line 

From
"ecmaVersion": 2020
To 
"ecmaVersion": 7

@mdjermanovic do you know why that is? is there something I should be worried about?

And thank you for your time!

**Comment by mdjermanovic** [comment]
> From
> "ecmaVersion": 2020
> To
> "ecmaVersion": 7

Hm, that shouldn't make a difference in this case, unless the errors you were expecting to see are parsing errors due to the use of some > ES7 syntax.

**Comment by JohnCasab** [comment]
Hi again @mdjermanovic, 

I thought I fixed it but it didn't help. Sometimes it's working and sometimes it's not. 

When I run: `yarn lint` after I ran `yarn lint --debug`

Then it's working. Before that it didnt work.

**Comment by JohnCasab** [comment]
I also get this warning when I run `yarn add yarn` command

`warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.`

**Comment by JohnCasab** [comment]
I also have many errors: 

```
✖ 416 problems (321 errors, 95 warnings)
  7 errors and 0 warnings potentially fixable with the `--fix` option.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Maybe thats the reason?

**Comment by mdjermanovic** [comment]
Can you run ESLint with a formatter that lists all files being linted, like [tap](https://eslint.org/docs/user-guide/formatters/#tap) (just add `-f tap` to the command).

Then we could see if this is a problem with not linting some files or not reporting the expected errors in those files.

**Comment by JohnCasab** [comment]
you mean yarn lint -f tap?

**Comment by mdjermanovic** [comment]
Can you redirect output to a file (e.g., add `> somefile.txt` to the end of the command) and then see if there's anything unexpected in the output, like a file that you were expecting to be linted not being in the list, or being in the list but there are no expected errors.

**Comment by JohnCasab** [comment]
yarn lint test.js
yarn run v1.22.10
$ ./node_modules/.bin/eslint '**/*.js' test.js

Oops! Something went wrong! :(

ESLint: 7.15.0

No files matching the pattern "test.js" were found.
Please check for typing mistakes in the pattern.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

test.js is a file inside .storybook folder, so im expecting it to be linted, but thats not happening.

**Comment by mdjermanovic** [comment]
> $ ./node_modules/.bin/eslint '**/*.js' test.js

You're probably getting this error since `./test.js` doesn't exist.

You can pass `.storybook/test.js`, or just omit it if it already matches the `**/*.js` pattern.

**Comment by mdjermanovic** [comment]
> When I run the command `yarn lint .storybook/test.js`
> I can see the errors for the test.js, BUT also all the other files that I want to be linted?
> 
> ```
> yarn lint .storybook/test.js
> yarn run v1.22.10
> $ ./node_modules/.bin/eslint '**/*.js' .storybook/test.js
> ```

`yarn lint .storybook/test.js` runs the `lint` script `"lint": "./node_modules/.bin/eslint '**/*.js'"` from your `package.json`, with an additional argument `.storybook/test.js`, so the command `yarn lint .storybook/test.js` gets translated to `./node_modules/.bin/eslint '**/*.js' .storybook/test.js`.

ESLint gets two patterns, `'**/*.js'` and `.storybook/test.js` and lints all files that match any of the patterns.

If you want to lint only `.storybook/test.js`, you can run `yarn eslint .storybook/test.js`.

**Comment by JohnCasab** [comment]
Alright thank you @mdjermanovic for your explanation. But so why is it that the command `yarn lint` only work sometimes?

**Comment by mdjermanovic** [comment]
> But so why is it that the command `yarn lint` only work sometimes?

I'm guessing there are too many errors so you can't see them all in your terminal. That would explain why `--fix` works on the files that don't seem to appear in the output without `--fix`.

Can you run `yarn lint > tmp.txt` and then open `tmp.txt` to see if there are all expected errors there?

**Comment by JohnCasab** [comment]
Yes I see all the errors in the tmp.txt file, so the problem was always that to many errors that the terminal could't display them. 

Thank you @mdjermanovic !

Should I be worried about the `error Command failed with exit code 1.`?

**Comment by mdjermanovic** [comment]
> Should I be worried about the `error Command failed with exit code 1.`?

No, that's expected behavior. When ESLint finds linting errors, exit code will be 1.

**Comment by nzakas** [comment]
Converting to a discussion as there doesn't appear to be any action item here
